### PR TITLE
Fix onnx backend to use GCS compatible API to load models

### PR DESCRIPTION
**Pull request by xprotobeast2**

### CODE REVIEW

File: src/backends/onnx/onnx_backend.cc [hunk 1]
```diff
@@ -75,17 +75,18 @@ OnnxBackend::Init(const std::string& path, const ModelConfig& config)
...
   // Create a "prototype" session option, which will be cloned and set
   // context-specific option on context creation.
-  OrtSessionOptions* session_options = OrtCreateSessionOptions();
-  OrtSetSessionThreadPoolSize(session_options, 1);
+  OrtSessionOptions* session_options;
+  RETURN_IF_ORT_ERROR(OrtCreateSessionOptions(&session_options));
+  RETURN_IF_ORT_ERROR(OrtSetSessionThreadPoolSize(session_options, 1));
   // disable graph optimization
-  OrtSetSessionGraphOptimizationLevel(session_options, 0);
+  RETURN_IF_ORT_ERROR(OrtSetSessionGraphOptimizationLevel(session_options, 0));
```
Comment (GuanLuo) [inline src/backends/onnx/onnx_backend.cc]: Need to be careful on making sure the Ort resource is released, we have to call its release function explicitly to do so.

This is going to be tedious, you can probably wrap the Ort resource inside a class and let the destructor calls the release function for you when the wrapped class object goes out of scope.

File: Dockerfile [hunk 2]
```diff
@@ -103,11 +103,12 @@ ARG ONNX_RUNTIME_VERSION=0.4.0
 # Get release version of Onnx Runtime
 WORKDIR /workspace
 RUN apt-get update && apt-get install -y --no-install-recommends git
-RUN git clone -b rel-${ONNX_RUNTIME_VERSION} --recursive https://github.com/Microsoft/onnxruntime
+RUN git clone --recursive https://github.com/Microsoft/onnxruntime
 
 ENV PATH="/opt/cmake/bin:${PATH}"
 ARG SCRIPT_DIR=/workspace/onnxruntime/tools/ci_build/github/linux/docker/scripts
-RUN ${SCRIPT_DIR}/install_ubuntu.sh && ${SCRIPT_DIR}/install_deps.sh
+RUN cp -r ${SCRIPT_DIR} /tmp/scripts && \
```
[the remark this answers is not in the document]
Comment (xprotobeast2) [reply Dockerfile]: It is used by install_deps.sh. The script expects SCRIPT_DIR to be mounted at /tmp/scripts.

File: Dockerfile [hunk 3]
```diff
@@ -103,11 +103,12 @@ ARG ONNX_RUNTIME_VERSION=0.4.0
 # Get release version of Onnx Runtime
 WORKDIR /workspace
 RUN apt-get update && apt-get install -y --no-install-recommends git
-RUN git clone -b rel-${ONNX_RUNTIME_VERSION} --recursive https://github.com/Microsoft/onnxruntime
```
[the remark this answers is not in the document]
Comment (xprotobeast2) [reply Dockerfile]: We have to work off of master until the next release. Release 0.4.0 does not have the API we require.

File: src/backends/onnx/loader.cc [hunk 4]
```diff
@@ -105,10 +105,9 @@ OnnxLoader::LoadSession(
     // Workaround for creating session on multiple GPUs until the issue
     // on Onnx Runtime is resolved
     // https://github.com/microsoft/onnxruntime/issues/1034
```
Comment (GuanLuo) [inline src/backends/onnx/loader.cc]: Can remove this comment

**Pull request merged**